File: Esercizio Esame (Customer)/src/Entity/CartProduct.php
<?php
    
    namespace Entity;
    
    class CartProduct
    {
        
        /**
         * @var int
         */
        private $id;
        /**
         * @var string
         */
        private $name;
        /**
         * @var string
         */
        private $photo_url;
        /**
         * @var float
         */
        private $unit_price;
        /**
         * @var int
         */
        private $qta;
        
        /**
         * @return int
         */
        public function getId(): int
        {
            return $this->id;
        }
        
        /**
         * @param int $id
         */
        public function setId(int $id): void
        {
            $this->id = $id;
        }
        
        /**
         * @return string
         */
        public function getName(): string
        {
            return $this->name;
        }
        
        /**
         * @param string $name
         */
        public function setName(string $name): void
        {
            $this->name = $name;
        }
        
        /**
         * @return string
         */
        public function getPhotoUrl(): string
        {
            return $this->photo_url;
        }
        
        /**
         * @param string $photo_url
         */
        public function setPhotoUrl(string $photo_url): void
        {
            $this->photo_url = $photo_url;
        }
        
        /**
         * @return float
         */
        public function getUnitPrice(): float
        {
            return $this->unit_price;
        }
        
        /**
         * @param float $unit_price
         */
        public function setUnitPrice(float $unit_price): void
        {
            $this->unit_price = $unit_price;
        }
        
        /**
         * @return int
         */
        public function getQta(): int
        {
            return $this->qta;
        }
        
        /**
         * @param int $qta
         */
        public function setQta(int $qta): void
        {
            $this->qta = $qta;
        }
        
    }

File: Esercizio Esame (Admin)/src/upo/gaborgalazzo/mweb/marketplace/domain/CartProduct.java
package upo.gaborgalazzo.mweb.marketplace.domain;

import upo.gaborgalazzo.mweb.marketplace.functiolanities.StringConverter;

import java.sql.ResultSet;
import java.sql.SQLException;

public class CartProduct
{
	private int id;
	private String name;
	private String photoUrl;
	private float unitPrice;
	private int qta;

	public static CartProduct fromResultSet(ResultSet resultSet) throws SQLException
	{
		CartProduct cartProduct = new CartProduct();
		cartProduct.setId(resultSet.getInt("id"));
		cartProduct.setQta(resultSet.getInt("qta"));
		cartProduct.setUnitPrice(resultSet.getFloat("unit_price"));
		cartProduct.setPhotoUrl(StringConverter.toUtf8(resultSet,"photo_url"));
		cartProduct.setName(StringConverter.toUtf8(resultSet,"name"));
		return cartProduct;
	}

	public int getId()
	{
		return id;
	}

	public void setId(int id)
	{
		this.id = id;
	}

	public String getName()
	{
		return name;
	}

	public void setName(String name)
	{
		this.name = name;
	}

	public String getPhotoUrl()
	{
		return photoUrl;
	}

	public void setPhotoUrl(String photoUrl)
	{
		this.photoUrl = photoUrl;
	}

	public float getUnitPrice()
	{
		return unitPrice;
	}

	public void setUnitPrice(float unitPrice)
	{
		this.unitPrice = unitPrice;
	}

	public int getQta()
	{
		return qta;
	}

	public void setQta(int qta)
	{
		this.qta = qta;
	}
}
